components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -391,7 +394,25 @@ private Response handleOAuthProblemException(OAuthMessage oAuthMessage, OAuthPro
 
         if (StringUtils.equals(oAuthMessage.getRequest().getParameter(RESPONSE_MODE), RESPONSE_MODE_FORM_POST)) {
             e.state(retrieveStateForErrorURL(oAuthMessage.getRequest(), oAuth2Parameters));
-            return Response.ok(createErrorFormPage(oAuthMessage.getRequest().getParameter(REDIRECT_URI), e)).build();
+            if (useOAuthResponseJspPage) {
+                try {
+                    HttpServletRequest request = oAuthMessage.getRequest();
+                    HttpServletResponse response = oAuthMessage.getResponse();
+                    String params = buildErrorParams(e);
+                    request.setAttribute("params", params);
+                    request.setAttribute("redirectURI", oAuthMessage.getRequest().getParameter(REDIRECT_URI));
+                    ServletContext authEndpoint = request.getServletContext().getContext("/authenticationendpoint");
+                    RequestDispatcher requestDispatcher = authEndpoint.getRequestDispatcher("/oauth_response.jsp");
```

Comment:
let's define a constant for these /authenticationendpoint and /oauth_response.jsp and they are used in more than one place.

Reply:
Fixed with fcba3c712827c2b8920198cb6431095679161835